index.bs

```diff
@@ -2129,11 +2189,108 @@ an [=attribution trigger=] |trigger|:
+
+To <dfn>obtain a null report</dfn> given an [=attribution trigger=] |trigger| and a [=moment=] |sourceTime|:
+
+1. Let |reportTime| be the result of running [=obtain an aggregatable report delivery time=] with |trigger|'s
+    [=attribution trigger/trigger time=].
+1. Let |contribution| be a new [=aggregatable contribution=] with the items:
+
+    : [=aggregatable contribution/key=]
+    :: A random non-negative 128-bit integer with uniform probability
+    : [=aggregatable contribution/value=]
+    :: 0
+1. Let |report| be a new [=aggregatable report=] struct whose items are:
+
+    : [=aggregatable report/reporting endpoint=]
+    :: |trigger|'s [=attribution trigger/reporting endpoint=]
+    : [=aggregatable report/effective attribution destination=]
+    :: |trigger|'s [=attribution trigger/attribution destination=]
+    : [=aggregatable report/source time=]
+    :: |sourceTime|
+    : [=aggregatable report/original report time=]
+    :: |reportTime|
+    : [=aggregatable report/report time=]
+    :: |reportTime|
+    : [=aggregatable report/report id=]
+    :: The result of [=generating a random UUID=]
+    : [=aggregatable report/source debug key=]
+    :: null
+    : [=aggregatable report/trigger debug key=]
+    :: |trigger|'s [=attribution trigger/debug key=]
+    : [=aggregatable report/contributions=]
+    :: « |contribution| »
+    : [=aggregatable report/serialized private state token=]
+    :: null
     : [=aggregatable report/aggregation coordinator=]
     :: |trigger|'s [=attribution trigger/aggregation coordinator=]
+    : [=aggregatable report/source registration time configuration=]
+    :: |trigger|'s [=attribution trigger/aggregatable source registration time configuration=]
+    : [=aggregatable report/is null report=]
+    :: true
 1. Return |report|.
 
+To <dfn>generate null reports</dfn> given an [=attribution trigger=] |trigger|:
+
+1. Let |nullReports| be a new [=list/is empty|empty=] [=list=].
+1. If |trigger|'s [=attribution trigger/aggregatable source registration time configuration=] is "<code>[=aggregatable source registration time configuration/exclude=]</code>":
+    1. Let |numNullReports| be the result of [=obtaining a randomized number of null reports=] with
+        [=randomized null report distribution parameter excluding source registration time=].
+    1. Repeat the following steps |numNullReports| times:
+        1. Let |nullReport| be the result of [=obtaining a null report=] with |trigger| and
+            |trigger|'s [=attribution trigger/trigger time=].
+        1. Add |nullReport| to the [=aggregatable report cache=].
+        1. [=list/Append=] |nullReport| to |nullReports|.
+    1. Return |nullReports|.
+1. Let |maxSourceExpiry| be [=max source expiry=].
+1. Round |maxSourceExpiry| away from zero to the nearest day (86400 seconds).
+1. For each integer |day| between 0 (inclusive) and the number of days in |maxSourceExpiry| (inclusive):
```

Comment:
> the number of days in |maxSourceExpiry|

This is a bit vague. Can we specify how to calculate that?

Reply:
Discussed offline, maxSourceExpiry is already rounded to the nearest day, so it's fine leaving as is.